src/util/colorize.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+Part of Minetest
+Copyright (C) 2024 rubenwardy
+
+Permission to use, copy, modify, and distribute this software for any
+purpose with or without fee is hereby granted, provided that the above
+copyright notice and this permission notice appear in all copies.
+
+THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+
+#include "colorize.h"
+
+#ifdef USE_CURL
+
+#include <curl/urlapi.h>
+#include "log.h"
+#include "string.h"
+#include <sstream>
+
+bool colorize_url(std::string &out, const std::string &url) {
+	// Forbid escape codes in URL
+	if (url.find('\x1b') != std::string::npos) {
+		out = "Unable to open URL as it contains escape codes";
+		return false;
+	}
+
+	CURLU *h = curl_url();
```

Comment:
you can probably turn this into a `std::unique_ptr<CURLU, curl_url_cleanup>` and have it deallocate automatically